src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -151,7 +158,10 @@ public void setEntityCallbacks(EntityCallbacks entityCallbacks) {
 			});
 		}
 
-		return entities;
+		return indexQueries.stream()
+				.map(IndexQuery::getObject)
```

Comment:
This must not be null, see my comment for the `maybeCallbackAfterSave(T entity)` method.

Reply:
Currently, this can only be null if some entity in the original Iterable is null.

Do you still suggest adding a non-null assertion?